include/net/buf.h

```diff
@@ -19,6 +19,12 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Network buffer library
+ * @defgroup net_buf Network Buffer Library
```

Comment:
Thanks for the info, I was kind of lost where to put the reference but should have digged deeper in the doc/ directories. I will send a new version that adds the required info there.